pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -113,6 +128,15 @@ func (c controller) List(typ config.GroupVersionKind, namespace string) ([]confi
 	input.Namespaces = namespaces
 	output := convertResources(input)
 
+	// Handle all status updates
+	// TODO we should probably place these on a queue using pilot/pkg/status
+	c.handleStatusUpdates(input.GatewayClass)
```

Comment:
Would it better if we define handleStatusUpdates on KubernetesResources struct and call it once - let it handle all the resources it knows of?